File: CPP/53.最大子序和.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        if(nums.size() == 0) return 0;
        int res = nums[0];
        int dp[nums.size()];
        dp[0] = nums[0];
        for(int i = 1; i < nums.size(); i++){
            dp[i] = max(dp[i-1] + nums[i], nums[i]);
            res = max(dp[i], res);
        }
        return res;
    }
};

File: Python/53.最大子序和.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        if n == 0:
            return 0
        #初始化n维dp数组
        dp = [0]*n
        dp[0] = nums[0]
        res = nums[0]
        i = 1
        while i < n:
            dp[i] = max(dp[i-1] + nums[i],nums[i])
            res = max(res,dp[i])
            i = i + 1
        return res